### PR TITLE
Raise a helpful error when a datatype cannot be found in a schema

### PR DESCRIPTION
**Pull request by GiaJordan**
For the linked issue, the error arose because the data type was not spelled according to the conventions of the class label in the schema. The error raised by `networkx`: `return self._nodes[n] \ KeyError: 'AssayRNASeqMetadataTemplate'` was not very good at communicating what was going wrong. I've wrapped another error around this message for the case of generating a manifest that should be more explanatory and offers suggestions for remedying.

Depends on: #1287

### CODE REVIEW

File: schematic/manifest/generator.py [hunk 1]
```diff
@@ -83,7 +83,16 @@ def __init__(
         # Determine whether current data type is file-based
         is_file_based = False
         if self.root:
-            is_file_based = "Filename" in self.sg.get_node_dependencies(self.root)
+            try:
```
Comment (mialy-defelice) [inline schematic/manifest/generator.py:86]: I would move this outside this particular call, looking for filename. Perhaps just upstream after you call the schemaGenerator You could call `is_class_in_schema` (might be `sg.se.is_class_in_schema` from the schema explorer. If it evaluates to false, you could raise the error and break.

**Review by mialy-defelice — CHANGES_REQUESTED** [review]

File: schematic/manifest/generator.py (hunk 1 above)
Comment (GiaJordan) [reply schematic/manifest/generator.py:86]: That makes sense!

**Comment by andrewelamb** [comment]
@GiaJordan Lets add a test to check that this results in the intended error.

**Comment by GiaJordan** [comment]
@andrewelamb thanks sounds good

**Comment by GiaJordan** [comment]
@andrewelamb I've added a test to ensure the exception gets raised in the correct circumstances. Could you please review when you have the time? Thanks!

**Review by andrewelamb — APPROVED** [review]

**Review by mialy-defelice — APPROVED** [review]

File: schematic/manifest/generator.py [hunk 2]
```diff
@@ -82,8 +82,19 @@ def __init__(
 
         # Determine whether current data type is file-based
         is_file_based = False
-        if self.root:
+
+        if self.root:
```
[the remark this answers is not in the document]
Comment (GiaJordan) [reply schematic/manifest/generator.py]: Okay so I've reworked the logic in that area, and also updated the tests to now check for this new `ValueError` as well. Could you take a look @mialy-defelice ?

[the remark this answers is not in the document]
Comment (GiaJordan) [reply schematic/manifest/generator.py]: So it seems like another test is actually checking to make sure that `None` is allowed to be passed in for `root`. It might be better to separate out this other change from the new error message @mialy-defelice
